ui/lib/replication/addon/templates/mode/secondaries/add.hbs

```diff
@@ -6,72 +6,74 @@
-      </div>
-      <p class="help has-text-grey">
-        This will be used to identify secondary cluster once a connection has been established with the primary.
-      </p>
-    </div>
-    <div class="field">
-      {{ttl-picker onChange=(action (mut ttl)) class="is-marginless"}}
-      <p class="help has-text-grey">
-        This is the Time To Live for the generated secondary token. After this period, the generated token will no longer be valid.
-      </p>
+    <p class="help has-text-grey">
+      This will be used to identify secondary cluster once a connection has been established with the primary.
+    </p>
+  </div>
+  <div class="field">
+    {{ttl-picker onChange=(action (mut ttl)) class="is-marginless"}}
+    <p class="help has-text-grey">
+      This is the Time To Live for the generated secondary token. After this period, the generated token will no longer be valid.
+    </p>
+  </div>
+  {{#if (eq replicationMode "performance")}}
+    <PathFilterConfigList
+      @paths={{paths}}
+      @config={{filterConfig}}
+      @id={{id}}
+    />
+  {{/if}}
+  <div class="field is-grouped box is-fullwidth is-bottomless">
+    <div class="control">
+      <button
+        type="submit"
+        class="button is-primary"
+        data-test-secondary-add=true
+        >
+        Generate token
+      </button>
     </div>
-    {{#if (eq replicationMode "performance")}}
-      <PathFilterConfigList
-        @paths={{paths}}
-        @config={{filterConfig}}
-        @id={{id}}
-      />
-    {{/if}}
-    <div class="field is-grouped box is-fullwidth is-bottomless">
-      <div class="control">
-        <button
-          type="submit"
-          class="button is-primary"
-          data-test-secondary-add=true
-          >
-          Generate token
-        </button>
-      </div>
-      <div class="control">
-        {{#link-to "mode.secondaries" replicationMode class="button"}}
-          Cancel
-        {{/link-to}}
-      </div>
+    <div class="control">
+      {{#link-to "mode.secondaries" replicationMode class="button"}}
+        Cancel
+      {{/link-to}}
     </div>
-  {{/if}}
+  </div>
 </form>
+
+{{#if isModalActive}}
+  <Modal @title="Copy your token" @onClose={{action "toggleModal" "Token copied!"}} @isActive={{isModalActive}}>
+    <section class="modal-card-body">
+      <p>This token can be used to enable {{replicationMode}} replication or change primaries on the secondary cluster.</p>
+      <div class="box is-shadowless is-fullwidth is-sideless">
+        <h2 class="title is-6">Activation token</h2>
+        <div class="copy-text level">
+          <div class="is-fullwidth">
+            <textarea readonly value={{token}} class="textarea level-left"/>
+          </div>
+        </div>
+        <div class="has-top-margin-xl has-bottom-margin-s">
+          {{info-table-row
+            label="TTL"
+            value=ttl}}
```

Comment:
is this value correct? i noticed when i entered a TTL of 60 seconds, the result displayed 60 minutes. not sure if this is a problem with the TTL picker or what but it's worth investigating:

![image](https://user-images.githubusercontent.com/903288/83817799-8f61e580-a67a-11ea-8014-9c29e5d348f5.png)

_![image](https://user-images.githubusercontent.com/903288/83817799-8f61e580-a67a-11ea-8014-9c29e5d348f5.png)_

Reply:
fixed with setting initalValue of ttl-picker.  This is the quick fix without replacing the old ttl-picker with the new one. That can be something we tackle in another, post project, ticket.